client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -18,7 +17,12 @@ import scala.annotation.tailrec
  *  }}}
  *  If an invalid path is passed to the constructor, it will return a failure.
  */
-case class PathBlob(protected val contents: HashMap[String, FileMetadata]) extends Dataset {
+case class PathBlob(
+  protected val contents: HashMap[String, FileMetadata],
+  protected val enableMDBVersioning: Boolean = false
+) extends Dataset {
+  override def prepareForUpload() = ???
```

Comment:
This will return an error. Is that the intention?

Reply:
Yes. I plan to implement one thing at a time. This PR is already way too big.